internal/rukpak/convert/registryv1.go

```diff
@@ -53,92 +53,73 @@ func RegistryV1ToPlain(rv1 fs.FS, installNamespace string, watchNamespaces []str
 	}
 	reg.PackageName = annotationsFile.Annotations.PackageName
 
-	var objects []*unstructured.Unstructured
 	const manifestsDir = "manifests"
-
-	entries, err := fs.ReadDir(rv1, manifestsDir)
-	if err != nil {
-		return nil, err
-	}
-	for _, e := range entries {
+	if err := fs.WalkDir(rv1, manifestsDir, func(path string, e fs.DirEntry, err error) error {
```

Comment:
There's another potential optimization here, where we spin up `runtime.NumCPU` goroutines and parse the files concurrently.  But that can happen as a follow-up.